### PR TITLE
[BRE-113] Publish SWIFT sdk releases to bitwarden/sdk-swift

### PR DESCRIPTION
**Pull request by michalchecinski**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/BRE-113

## 📔 Objective

Automate publishing SWIFT SDK to [bitwarden/sdk-swift](https://github.com/bitwarden/sdk-swift) repo. Thanks @mimartin12 for the initial work on that!

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/build-swift.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+name: Build Swift Package
+
+on:
+  push:
```
Comment (michalchecinski) [inline .github/workflows/build-swift.yml:5]: @Hinton Can we trigger only on changes in the `languages/swift` repository? Or should we trigger on every change in the repo?

Comment (Hinton) [reply .github/workflows/build-swift.yml:5]: It should be every change. I don't expect changes to languages/swift to be common.

File: .github/workflows/build-swift.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    outputs:
+      package_version: ${{ steps.retrieve-version.outputs.package_version }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+      - name: Get Package Version
+        id: retrieve-version
+        run: |
+          VERSION=$(grep -o '^version = ".*"' crates/bws/Cargo.toml | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+")
```
Comment (Hinton) [inline .github/workflows/build-swift.yml:24]: This should be the bitwarden crate not the cli.

Comment (michalchecinski) [reply .github/workflows/build-swift.yml:24]: Whoops.... Fixed 😃

Comment (Hinton) [reply .github/workflows/build-swift.yml]: @michalchecinski it occured to me know that the bitwarden verison is inherited from the workspace version. https://github.com/bitwarden/sdk/blob/main/Cargo.toml#L8

Comment (michalchecinski) [reply .github/workflows/build-swift.yml]: Got it. Also, should the version for SWIFT SDK be the same as Rust crates?

Comment (Hinton) [reply .github/workflows/build-swift.yml]: I think we can use that for now.

File: .github/workflows/release-swift.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            url: "https://github.com/bitwarden/sdk-swift/releases/download/v${{ env._RELEASE_NAME }}/BitwardenFFI-${{ env._PKG_VERSION }}-${{ needs.validate.outputs.short_sha }}.xcframework.zip",|' sdk/languages/swift/Package.swift
+
+          # Run swiftformat
+          swiftformat sdk/languages/swift/Package.swift
+
+          # Copy files to local sdk-swift repo path
+          cp --verbose -rf sdk/languages/swift/README.md sdk-swift/README.md
+          cp --verbose -rf sdk/languages/swift/Package.swift sdk-swift/Package.swift
+          cp --verbose -rf sdk/languages/swift/Sources sdk-swift/Sources
+          cp --verbose -rf sdk/languages/swift/Tests sdk-swift/Tests
```
Comment (Hinton) [inline .github/workflows/release-swift.yml:156]: Are we calculating the checksum somewhere?'

I usually have to edit these two lines
```
            url: "https://bwlivefronttest.blob.core.windows.net/sdk/de3b6e2-BitwardenFFI.xcframework.zip",
            checksum: "37e884d820907af85e788b5e0ab483bfc5aa0299e5d6b39865ad70659322a202"),
```

Comment (michalchecinski) [reply .github/workflows/release-swift.yml]: I thought I did it... It's fixed now. :)

**Review by vgrassia — DISMISSED** [review]
LGTM, please wait for Oscar to review.

File: .github/workflows/release-swift.yml [hunk 4]
```diff
@@ -84,11 +84,16 @@ jobs:
...
-            echo "release_name=${{ env._PKG_VERSION }}-unstable-${{ needs.validate.outputs.short_sha }}" >> $GITHUB_OUTPUT
+            echo "release_name=${{ steps.version.outputs.version }}-unstable-${{ steps.set-sha.outputs.short_sha }}" >> $GITHUB_OUTPUT
           else
-            echo "release_name=${{ env._PKG_VERSION }}" >> $GITHUB_OUTPUT
+            echo "release_name=${{ steps.version.outputs.version }}" >> $GITHUB_OUTPUT
           fi
 
+      - name: Calculate swift file checksum
+        run: |
+          CHECKSUM=$(swift package compute-checksum BitwardenFFI-${{ steps.version.outputs.version }}-${{ steps.set-sha.outputs.short_sha }}.xcframework)
```
Comment (Hinton) [inline .github/workflows/release-swift.yml:94]: Is this the hash of the zip file? I always run the compute-checksum on the already zipped framework.

Comment (michalchecinski) [reply .github/workflows/release-swift.yml:94]: Okay, I thought that needs to be on pre-zipped file. I'll change that.

**Review by Hinton — DISMISSED** [review]

File: .github/workflows/release-swift.yml [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      - name: Set release name
+        id: set-release-name
+        run: |
+          if [[ ${{ inputs.pre-release }} == true ]]; then
+            echo "release_name=${{ steps.version.outputs.version }}-unstable-${{ steps.set-sha.outputs.short_sha }}" >> $GITHUB_OUTPUT
+          else
+            echo "release_name=${{ steps.version.outputs.version }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Calculate swift file checksum
```
Comment (vgrassia) [inline .github/workflows/release-swift.yml:93]:
```suggestion
      - name: Calculate swift file checksum
        id: calculate-swift-checksum
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Review by vgrassia — APPROVED** [review]

**Pull request merged**